File: Pattern Problems/Triangle.java
/*
* 
* *
* * * 
* * * *
* * * * *
*/

public static void triangle(int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j <= i; j++) {
            System.out.print("*" + " ");
        }
        System.out.println();
    }
}

/*
* * * * * * * 
* * * * * *   
* * * * *     
* * * *       
* * *
* *
*
*/

public static void pattern(int n) {
    for (int i = n; i > 0; i--) {
        for (int j = 0; j < i; j++) {
            System.out.print("*" + " ");
        }
        System.out.println();
    }
}

/*
    *
   **
  ***
 ****
*****
*/

public static void pattern(int n) {
    for (int i = 5; i > 0; i--) {
        for (int j = 0; j < i - 1; j++) {
            System.out.print(" ");
        }
        for (int j = 0; j <= 5 - i; j++) {
            System.out.print("*");
        }
        System.out.println();
    }
}

/*
*****
 ****
  ***
   **
    *
*/

public static void pattern(int n) {
    for (int i = n; i > 0; i--) {
        for (int j = 0; j < n - i; j++) {
            System.out.print(" ");
        }
        for (int j = 0; j < i; j++) {
            System.out.print("*");
        }
        System.out.println();
    }
}

/*
* * * * * 
 * * * *  
  * * *   
   * *    
    * 
*/

public static void pattern(int n) {
    for (int i = n; i > 0; i--) {
        for (int j = 0; j < n - i; j++) {
            System.out.print(" ");
        }
        for (int j = 0; j < i; j++) {
            System.out.print("*" + " ");
        }
        System.out.println();
    }
}

File: Pattern Problems/Triangle.js
/*
* 
* *
* * * 
* * * *
* * * * *
*/

function pattern(n) {
    for (let i = 0; i < n; i++) {
        for (let j = 0; j <= i; j++) {
            process.stdout.write("*" + " ");
        }
        console.log();
    }
}

/*
* * * * * * * 
* * * * * *   
* * * * *     
* * * *       
* * *
* *
*
*/

function pattern(n) {
    for (let i = n; i > 0; i--) {
        for (let j = 0; j < i; j++) {
            process.stdout.write("*" + " ");
        }
        console.log();
    }
}

/*
    *
   **
  ***
 ****
*****
*/

function pattern(n) {
    for (let i = n; i > 0; i--) {
        for (let j = 0; j < i - 1; j++) {
            process.stdout.write(" ");
        }
        for (let j = 0; j <= n - i; j++) {
            process.stdout.write("*");
        }
        console.log();
    }
}

/*
    * 
   * *
  * * *
 * * * *
* * * * *
*/

function pattern(n) {
    for (let i = n; i > 0; i--) {
        for (let j = 0; j < i - 1; j++) {
            process.stdout.write(" ");
        }
        for (let j = 0; j <= n - i; j++) {
            process.stdout.write("*" + " ");      // just add space in the above program
        }
        console.log();
    }
}

/*
*****
 ****
  ***
   **
    *
*/

function pattern(n) {
    for (let i = n; i > 0; i--) {
        for (let j = 0; j < n - i; j++) {
            process.stdout.write(" ");
        }
        for (let j = 0; j < i; j++) {
            process.stdout.write("*");
        }
        console.log();
    }
}

/*
* * * * * 
 * * * *  
  * * *   
   * *    
    * 
*/

function pattern(n) {
    for (let i = n; i > 0; i--) {
        for (let j = 0; j < n - i; j++) {
            process.stdout.write(" ");
        }
        for (let j = 0; j < i; j++) {
            process.stdout.write("*" + " ");       // just add space in the above program
        }
        console.log();
    }
}
